src/types.rs

```diff
@@ -12,7 +12,7 @@ use crate::{
 macro_rules! typed {
     ($($kind:expr => $name:ident$(: $trait:ident)*$(: { $($block:tt)* })*),*) => {
         $(
-            #[derive(Clone, Debug)]
+            #[derive(Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
This was necessary to use the derive for `PartialEq, Eq, Hash` for `StrPart`.

Reply:
LGTM :+1: